File: client/src/app/models/ticket-card.ts
export interface TicketCard {
  conversationId: string
  conversationType: string | null
  owner: string | null

  ticketId: string
  subject: string
  status: string

  contactId: string
  contactDisplayName: string
  contactPhone: string | null
  contactProfilePic: string | null

  tags: TagItem[]

  lastMessageContent: string | null
  lastMessageCreatedAt: string | null

  read: boolean
}

export interface TagItem {
  id: string
  name: string
  colorCode: string
}

File: server/src/Models/Database/TicketCard.cs
using System.Text.Json.Serialization;

public class TicketCard
{
    public Guid ConversationId { get; set; }
    public string? ConversationType { get; set; }

    public Guid? Owner { get; set; }

    public Guid TicketId { get; set; }
    public string Subject { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;

    public Guid ContactId { get; set; }
    public Guid? ContactOwnerId { get; set; }
    public string ContactDisplayName { get; set; } = string.Empty;
    public string? ContactPhone { get; set; }
    public string? ContactProfilePic { get; set; }

    public TagItem[] Tags { get; set; } = [];

    public string? LastMessageContent { get; set; }
    public DateTime? LastMessageCreatedAt { get; set; }

    public bool Read { get; set; }
}

public class TagItem
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("colorCode")]
    public string ColorCode { get; set; } = string.Empty;
}
